### PR TITLE
Doc examples

### PR DESCRIPTION
**Pull request by Luthaf**
Add examples to functions in the `types` module.

Ref #21

### CODE REVIEW

**Comment by g-bauer** [comment]
Nice work.

I found some minor language things. Do you want me to comment in the code? Else I can go over the merged files ...

**Comment by Luthaf** [comment]
You can comment here, that should be easier.

File: src/core/src/types/arrays.rs [hunk 1]
```diff
@@ -5,42 +5,91 @@
...
 use types::Zero;
 
-/// Two dimmensional array, based on ndarray.
+/// Two dimmensional tensors, based on ndarray.
 ///
 /// Most of the methods are simply forwarded to ndarray, so also look the doc
-/// for this crate.
-#[derive(Debug, Clone)]
+/// for this crate. This array type mainly support indexing using tuples as
+/// indices and is though as storage backend for multi-dimmensional data.
```
Comment (g-bauer) [inline src/core/src/types/arrays.rs]: di**m**ensional

File: src/core/src/types/arrays.rs [hunk 2]
```diff
@@ -5,42 +5,91 @@
...
+    /// # use lumol::Array2;
+    /// let mut a = Array2::zeros((8, 5));
+    ///
+    /// a[(3, 3)] = 42.0;
+    ///
+    /// // This does nothing
+    /// a.resize_if_different((8, 5));
+    /// assert_eq!(a[(3, 3)], 42.0);
+    ///
+    /// // This allocate a new array
```
Comment (g-bauer) [inline src/core/src/types/arrays.rs]: allocate**s**

File: src/core/src/types/arrays.rs [hunk 3]
```diff
@@ -5,42 +5,91 @@
...
-impl<T> Array2<T> {
-    /// Get the shape of the array
-    pub fn shape(&self) -> (Ix, Ix) {
-        let shape = self.0.shape();
-        (shape[0], shape[1])
-    }
-}
-
 impl<T: Clone> Array2<T> {
     /// Assign the given scalar to all entry in this array
```
Comment (g-bauer) [inline src/core/src/types/arrays.rs]: entr**ies**

File: src/core/src/types/arrays.rs [hunk 4]
```diff
@@ -81,37 +134,79 @@ impl<T> IndexMut<(Ix, Ix)> for Array2<T> {
 
 /******************************************************************************/
 
-/// Three dimmensional array, based on ndarray
+/// Three dimmensional tensors, based on ndarray
 ///
 /// Most of the methods are simply forwarded to ndarray, so also look the doc
-/// for this crate.
-#[derive(Debug, Clone)]
+/// for this crate. This array type mainly support indexing using tuples as
```
Comment (g-bauer) [inline src/core/src/types/arrays.rs]: support**s**

File: src/core/src/types/arrays.rs [hunk 5]
```diff
@@ -81,37 +134,79 @@ impl<T> IndexMut<(Ix, Ix)> for Array2<T> {
...
+/// indices and is though as storage backend for multi-dimmensional data.
```
Comment (g-bauer) [inline src/core/src/types/arrays.rs]: di**m**ensional

File: src/core/src/types/complex.rs [hunk 6]
```diff
@@ -8,7 +8,35 @@ use std::f64;
...
+/// # use lumol::Complex;
+///
+/// let w = Complex::cartesian(-1.0, 0.5);
+/// let z = Complex::cartesian(4.0, 2.0);
+///
+/// // Addition
+/// let c = w + z;
+/// assert_eq!(c, Complex::cartesian(3.0, 2.5));
+///
+/// // Substraction
```
Comment (g-bauer) [inline src/core/src/types/complex.rs]: Su**bt**raction

File: src/core/src/types/matrix.rs [hunk 7]
```diff
@@ -6,14 +6,81 @@ use std::ops::{Add, Sub, Mul, Div, Index, IndexMut};
...
+///
+/// // Addition
+/// let c = a + one;
+/// assert_eq!(c, Matrix3::new(
+///     2.0, 0.0, 3.0,
+///     0.0, 3.0, 5.6,
+///     0.0, 0.0, 9.0
+/// ));
+///
+/// // Substraction
```
Comment (g-bauer) [inline src/core/src/types/matrix.rs]: Subtraction

**Pull request merged**